src/main/xsl/preprocess/maplinkImpl.xsl

```diff
@@ -340,7 +340,7 @@ See the accompanying LICENSE file for applicable license.
     </xsl:variable>
     
     <xsl:for-each select="ancestor::*[contains(@class, ' map/relrow ')]/*[contains(@class, ' map/relcell ')][position()!=$position]">
-      <xsl:if test="descendant::*[contains(@class, ' map/topicref ')][@href and not(@href = '')][not(@linking = ('none', 'sourceonly'))]">
+      <xsl:if test="descendant::*[contains(@class, ' map/topicref ')][(@href and not(@href = '')) or */*[contains(@class,' map/linktext ') or contains(@class,' topic/linktext ')]][not(@linking = ('none', 'sourceonly'))]">
```

Comment:
The added set of predicate conditions are repeated in multiple placed in this stylesheet. IMO this should be extracted into a function with a descriptive name to make the code easier to read and understand.

Reply:
Made the `[(href and not href='') or linktext]` condition into a function.

Still need to add an integration test for this to ensure the links are updated properly by `maplink` + not dropped by `topicpull`.